tests/e2e.rs

```diff
@@ -661,29 +695,110 @@ async fn test_e2e_sync_deactivated_only() {
 	}
 }
 
+#[test(tokio::test)]
+#[test_log(default_log_filter = "debug")]
+async fn test_e2e_ukt_sync() {
+	let mock_server = MockServer::start().await;
+
+	prepare_oauth2_mock(&mock_server).await;
+	prepare_endpoint_mock(&mock_server).await;
+
+	let mut config = config().await.clone();
+	clean_config_sources(&mut config, "ukt");
+
+	config
+		.sources
+		.ukt
+		.as_mut()
+		.map(|ukt| {
+			ukt.oauth2_url = get_mock_server_url(&mock_server, OATH2_PATH)
+				.expect("Failed to get mock server URL");
+			ukt.endpoint_url = get_mock_server_url(&mock_server, ENDPOINT_PATH)
+				.expect("Failed to get mock server URL");
+		})
+		.expect("UKT configuration is missing");
+
+	let user = ImportHumanUserRequest {
+		user_name: "first_email@example.com".to_owned(),
+		profile: Some(Profile {
+			first_name: "First".to_owned(),
+			last_name: "Last".to_owned(),
+			display_name: "First Last".to_owned(),
+			gender: Gender::Unspecified.into(),
+			nick_name: "nickname".to_owned(),
+			preferred_language: String::default(),
+		}),
+		email: Some(Email { email: "first_email@example.com".to_owned(), is_email_verified: true }),
+		phone: Some(Phone { phone: "+12015551111".to_owned(), is_phone_verified: true }),
+		password: String::default(),
+		hashed_password: None,
+		password_change_required: false,
+		request_passwordless_registration: false,
+		otp_code: String::default(),
+		idps: vec![],
+	};
+
+	let zitadel = open_zitadel_connection().await;
+	zitadel
+		.create_human_user(&config.zitadel.organization_id, user)
+		.await
+		.expect("failed to create user");
+
+	let user = zitadel
+		.get_user_by_login_name("first_email@example.com")
+		.await
+		.expect("could not query Zitadel users");
+	assert!(user.is_some());
+	let user = user.expect("could not find user");
+	assert_eq!(user.user_name, "first_email@example.com");
+
+	config.perform_sync().await.expect("syncing failed");
+
+	let user = zitadel.get_user_by_login_name("first_email@example.com").await;
+	assert!(user.is_err_and(|error| matches!(error, ZitadelError::TonicResponseError(status) if status.code() == TonicErrorCode::NotFound)));
+}
+
+/// Get rid of unwanted sources in the configuration
```

Comment:
Hmm, could we just have multiple template files, maybe using the yaml import features if we really need to DRY the zitadel config and feature flags? This is a little too magic IMO.

Reply:
In addition, we should probably make a new test group for the ukt tests, and not run the same test setup script for both. That can probably done in a separate PR, since it doesn't hurt for now.

Reply:
We don't need to remove the sources from the config. That was just a bonus to make things cleaner, but since it's not welcome, I will simply delete it. The e2e tests use one config. The unit tests for separate sources and config use their own versions.

Agreed, additional splitting of test groups etc can be done in a separate PR if deemed necessary.